testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java

```diff
@@ -197,7 +201,25 @@ public Object[][] parallelTests()
     @Test(invocationCount = INVOCATION_COUNT, dataProvider = "parallelTests")
     public final void testParallel(Runnable runnable)
     {
-        runnable.run();
+        try {
+            // currently it is not possible to limit number of threads to be used for executing tests
+            // generated by "parallel" data provider. Using semaphore as a stop-gap. Default parallelism of
+            // 10 turned out to consume too much resources.
```

Comment:
We have no plans to shift that limitations so let's word this a bit differently

```
By default, a test method using a @DataProvider with parallel attribute is run in 10 threads (org.testng.xml.XmlSuite#DEFAULT_DATA_PROVIDER_THREAD_COUNT).
Limit number of concurrent test executions.
```

you do not need to say that "currently it is not possible to limit number of threads", because  that's obvious. If there was any better way to do this, you'd do it differently.